### PR TITLE
[query][qob][hailtop/fs] Expose modification and creation time.

### LINKED ISSUE #12540: hl.hadoop_stat no longer works for requester pays bucket
After updating from hail from v0.2.92 to v0.2.105 , I've started getting the "Bucket is a requester pays bucket but no user project provided" error  (with further details in  https://hail.zulipchat.com/#narrow/stream/123010-Hail-Query-0.2E2-support/topic/hl.2Ehadoop_stat.20no.20longer.20works.20for.20requester.20pays.20bucket)

### PR DESCRIPTION
**Pull request by danking**
Fixes #12540.

CHANGELOG: When using Query-on-Batch, hl.hadoop* methods now properly support creation and modification time.

Creation time is supported by modern Linuxes but only through a new statx API which is not exposed by the Python standard library. There is a 0.1 version library from 2021 which exposes statx including the "birth time". I chose to raise an exception for now.

Each cloud does support a "modification time" but it generally refers to changes to metadata or is just the creation time:
- https://cloud.google.com/storage/docs/json_api/v1/objects#resource (see updated)
- https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html#API_GetObject_ResponseSyntax ("Last-Modified", is always creation time)
- https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#API_HeadObject_ResponseSyntax same as above
- https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob (see Last-Modified and x-ms-creation-time)

### CODE REVIEW

**Comment by danking** [comment]
I also just implemented RC3339 parsing because Google's timestamps are not compliant with ISO8601.

File: hail/python/hail/fs/router_fs.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def name(self):
 
     def close(self):
         self.aws.close()
-        async_to_blocking(self.cm.__aexit__())
+        async_to_blocking(self.cm.__aexit__(None, None, None))
```
Comment (danking) [inline hail/python/hail/fs/router_fs.py:104]: Random other bug I found while I was in here.

**Comment by danking** [comment]
tests and lints resolved

**Review by jigold — CHANGES_REQUESTED** [review]

File: hail/python/test/hailtop/inter_cloud/test_fs.py [hunk 2]
```diff
@@ -472,6 +473,33 @@ async def test_statfile(filesystem: Tuple[asyncio.Semaphore, AsyncFS, str]):
...
+    status = await fs.statfile(file)
+
+    if isinstance(fs, RouterAsyncFS):
+        is_local = isinstance(fs._get_fs(file), LocalAsyncFS)
+    else:
+        is_local = isinstance(fs, LocalAsyncFS)
+
+    if is_local:
+        try:
+            status.time_created()
```
[the remark this answers is not in the document]
Comment (danking) [reply hail/python/test/hailtop/inter_cloud/test_fs.py:493]: good catch

**Review by danking — COMMENTED** [review]
addressed

**Review by jigold — APPROVED** [review]

**Pull request merged**